upup/pkg/fi/nodeup/nodetasks/package.go

```diff
@@ -293,16 +301,26 @@ func (_ *Package) RenderLocal(t *local.LocalTarget, a, e, changes *Package) erro
 			}
 
 			var args []string
+			env := os.Environ()
 			if t.HasTag(tags.TagOSFamilyDebian) {
-				args = []string{"dpkg", "-i"}
+				// Only Debian releases newer than Jessie can install .deb via apt-get
+				// TODO: Refactor this function when Jessie support is dropped (duplicated code)
+				if t.HasTag("_jessie") {
+					args = []string{"dpkg", "-i"}
+				} else {
+					args = []string{"apt-get", "install", "--yes", "--no-install-recommends"}
+					env = append(env, "DEBIAN_FRONTEND=noninteractive")
+				}
 			} else if t.HasTag(tags.TagOSFamilyRHEL) {
-				args = []string{"/usr/bin/rpm", "-i"}
+				args = []string{"/usr/bin/yum", "install", "-y"}
```

Comment:
I think i'm on board with `apt-get install -y --no-install-recommends` replacing `dpkg -i`.
I haven't been in rhel world for awhile. is `yum install -y` the correct/equivalent of the apt-get command above? or is there a way to restrict installing the suggestions?

Reply:
RPM specs allows weak dependencies, but it's a newer addition compared to Debian.
RHEL/CentOS 7 has no such flag to not install recommended packages, but I am not sure how widely used is there.
RHEL/CentOS 8 has switched to DNF (yum is a link to DNF) and has `--setopt=install_weak_deps=False`.
I will add a special case here for RHEL/CentOS 8 just to be safe.

Reply:
```bash
~ cat /etc/centos-release
CentOS Linux release 8.0.1905 (Core) 

~ /usr/bin/dnf install -y --setopt=install_weak_deps=False *.rpm
Last metadata expiration check: 0:06:42 ago on Tue 17 Dec 2019 07:24:34 AM UTC.
Dependencies resolved.
==============================================================================================================================================================================================================================================================
 Package                                                           Arch                                        Version                                                                                Repository                                         Size
==============================================================================================================================================================================================================================================================
Installing:
 containerd.io                                                     x86_64                                      1.2.10-3.2.el7                                                                         @commandline                                       23 M
 docker-ce                                                         x86_64                                      3:19.03.4-3.el7                                                                        @commandline                                       24 M
 docker-ce-cli                                                     x86_64                                      1:19.03.4-3.el7                                                                        @commandline                                       39 M
Installing dependencies:
 container-selinux                                                 noarch                                      2:2.94-1.git1e99f1d.module_el8.0.0+58+91b614e7                                         AppStream                                          43 k
 checkpolicy                                                       x86_64                                      2.8-2.el8                                                                              BaseOS                                            338 k
 libcgroup                                                         x86_64                                      0.41-19.el8                                                                            BaseOS                                             70 k
 policycoreutils-python-utils                                      noarch                                      2.8-16.1.el8                                                                           BaseOS                                            228 k
 python3-audit                                                     x86_64                                      3.0-0.10.20180831git0047a6c.el8                                                        BaseOS                                             85 k
 python3-libsemanage                                               x86_64                                      2.8-5.el8                                                                              BaseOS                                            127 k
 python3-policycoreutils                                           noarch                                      2.8-16.1.el8                                                                           BaseOS                                            2.2 M
 python3-setools                                                   x86_64                                      4.2.0-2.el8                                                                            BaseOS                                            598 k
Enabling module streams:
 container-tools                                                                                               rhel8                                                                                                                                         

Transaction Summary
==============================================================================================================================================================================================================================================================
Install  11 Packages
```